Source/santad/EventProviders/FAAPolicyProcessor.mm

```diff
@@ -0,0 +1,128 @@
+/// Copyright 2025 North Pole Security, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#include "Source/santad/EventProviders/FAAPolicyProcessor.h"
+
+#import "Source/common/MOLCertificate.h"
+#import "Source/common/MOLCodesignChecker.h"
+
+// Terminal value that will never match a valid cert hash.
+NSString *const kBadCertHash = @"BAD_CERT_HASH";
+
+namespace santa {
+
+FAAPolicyProcessor::FAAPolicyProcessor(SNTDecisionCache *decision_cache)
+    : decision_cache_(decision_cache) {}
+
+NSString *FAAPolicyProcessor::GetCertificateHash(const es_file_t *es_file) {
+  // First see if we've already cached this value
+  NSLog(@"GetCertificateHash ENTER");
+  SantaVnode vnodeID = SantaVnode::VnodeForFile(es_file);
+  NSString *result = cert_hash_cache_.get(vnodeID);
+  if (!result) {
+    // If this wasn't already cached, try finding a cached SNTCachedDecision
+    SNTCachedDecision *cd = [decision_cache_ cachedDecisionForFile:es_file->stat];
+    if (cd) {
+      // There was an existing cached decision, use its cert hash
+      result = cd.certSHA256;
+    } else {
+      // If the cached decision didn't exist, try a manual lookup
+      NSError *e;
+      MOLCodesignChecker *csInfo =
+          [[MOLCodesignChecker alloc] initWithBinaryPath:@(es_file->path.data) error:&e];
+      if (!e) {
+        result = csInfo.leafCertificate.SHA256;
+      }
```

Comment:
You could use the non-error variant here, given that you don't care about the error:

```
MOLCodesignChecker *csInfo = [[MOLCodesignChecker alloc] initWithBinaryPath:@(es_file->path.data)];
result = csInfo.leafCertificate.SHA256;
```